### DISCUSSION TITLE
How run setxkbmap with Xwayland?

### DISCUSSION THREAD

**Discussion by LinArcX** [WayfireWM/wayfire#1542] [Q&A] [answered]
Hello.

I recently switch to Wayland after a long time that I was using X. And wayfire is my first touch with this world!

One of my problems is that if I run `setxkbmap` from a terminal, I'll get this error message:

↳ setxkbmap WARNING: Running setxkbmap against an XWayland server

And if I run it with Xwayland:
```
(EE)
Fatal server error:
(EE) Unrecognized option: setxkbmap
(EE)
```
How can I run it with Xwayland?

**Comment by soreau** [comment]
You want to put the setxkbmap options in input.xkb_options

**Reply by LinArcX** [reply]
I tried this:

```
[input]
xkb_layout = us,ir
xkb_variant = qwerty,qwerty
xkb_options = -model pc105 -layout us,ir -option grp:alt_shift_toggle
```

But toggling with Shift+Alt won't work.

This is what I did in X:
`setxkbmap -model pc105 -layout us,ir -option grp:alt_shift_toggle`

Edit: Even I tried this, but still, it's not working:
`xkb_options = grp:alt_shift_toggle`

**Comment by LinArcX** [comment, marked as answer]
This is working:

```
[input]
xkb_layout = us,ir
xkb_variant = qwerty
xkb_model = pc105
xkb_options = grp:alt_shift_toggle
natural_scroll = false
kb_numlock_default_state = true
```

**Reply by IAmADeviler** [reply]
Where I have to paste this??

**Reply by qwuser234** [reply]
in wafire.ini config lol
